nxdrive/client/local/linux.py

```diff
@@ -24,8 +26,19 @@ class LocalClient(LocalClientMixin):
 
     def has_folder_icon(self, ref: Path) -> bool:
         """Check if the folder icon is set."""
-        # To be implementation with https://jira.nuxeo.com/browse/NXDRIVE-1831
-        return True
+        matcher = re.compile(r"metadata::emblems: \[nuxeo.*\]")
+        folder_path = self.abspath(ref).as_posix()
+        try:
+            output = subprocess.check_output(
+                ["gio", "info", "-a", "metadata", folder_path], encoding="utf-8"
+            )
+        except subprocess.CalledProcessError:
+            log.debug(f"Could not check the metadata of {ref!r}")
+            return False
+        result = matcher.findall(output)
```

Comment:
You could move `matcher = ...` at this line, as it will be useless if the `subprocess` call fails.
Also, you can refactor those lines:
```suggestion
        return bool(matcher.findall(output))
```

(and add a empty line before the whole block to make clear this is not the same "action" being done)